### PR TITLE
feat: Added multithreading execution of data loading

### PR DESCRIPTION
**Pull request by fg-mindee**
Following up on #198, this PR introduces the following modifications:
- adds a multithreading mapping function
- adds `workers` constructor arguments to dataloaders to enable multithreading
- added unittests
- updated training scripts

Any feedback is welcome!

### CODE REVIEW

File: doctr/datasets/multithreading.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import multiprocessing as mp
+from multiprocessing.pool import ThreadPool
+from typing import Callable, Any, Optional, Iterable
+
+
+__all__ = ['multithread_exec']
+
+
+def multithread_exec(func: Callable[[Any], Any], seq: Iterable[Any], threads: Optional[int] = None) -> Iterable[Any]:
+    """Download a file accessible via URL with mutiple retries
```
Comment (charlesmindee) [inline doctr/datasets/multithreading.py:16]: I think this is not appropriated to this function

**Comment by charlesmindee** [comment]
Thanks for the PR! Only a small comment on a docstring

File: doctr/datasets/multithreading.py (hunk 1 above)
Comment (fg-mindee) [reply doctr/datasets/multithreading.py]: Good catch!

**Comment by charlesmindee** [comment]
Thanks!

**Review by charlesmindee — APPROVED** [review]
Thanks!

**Pull request merged**